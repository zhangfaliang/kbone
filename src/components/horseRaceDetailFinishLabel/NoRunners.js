import React from 'react';
import styles from './index.less';

const NoRunners = ({ prefixCls = 'default', noRunText = '', noRunners = [''] }) => {
  return (
    <div className={styles[`${prefixCls}-no-runners`]}>
      <span className={styles.noRunText}>{noRunText}</span>
      {noRunners.map(noRunner => (
        <span key={noRunner} className={styles.noRunner}>
          {noRunner}
        </span>
      ))}
    </div>
  );
};

NoRunners.COMPONENT_NAME = 'NORUNNERS';

export default NoRunners;
