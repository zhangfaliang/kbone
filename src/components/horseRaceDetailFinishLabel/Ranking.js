import React from 'react';
import styles from './index.less';

const Ranking = ({ prefixCls = 'default', rankingText = '1st' }) => {
  return <span className={styles[`${prefixCls}-ranking`]}>{rankingText}</span>;
};

export default Ranking;
