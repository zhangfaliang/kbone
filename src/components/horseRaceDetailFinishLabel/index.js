export { default as FinishLabel } from './FinishLabel';
export { default as Ranking } from './Ranking';
export { default as NoRunners } from './NoRunners';
