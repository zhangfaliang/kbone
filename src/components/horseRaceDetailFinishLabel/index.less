@import '../../theme/index.less';

.finish-label {
  height: 100px;
  width: 710px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
  box-sizing: border-box;
  border-top: 1px solid @bet-border-color;
  background: #FFFFFF;

  .left {
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: Akrobat-SemiBold;
    font-size: 32px;
    color: @primary-font-color;
  }

  .center {
    font-family: Akrobat-Regular;
    font-size: 32px;
    color: @progress-text-color;
  }

  .right {
    display: flex;
    justify-content: center;
    align-items: flex-end;
    flex-flow: column nowrap;
    font-family: Akrobat-Regular;
    font-size: 32px;
    color: @progress-text-color;

    .noEntry {
      font-family: Akrobat-Regular;
      font-size: 32px;
      color: @aide-color-secondary;
    }
  }
}
.close-label{
  &:extend(.finish-label all);
}
.fore-tri-finish-label{
  &:extend(.finish-label all);
  .left,.center,.right{
    width: 33.33%;
    display: flex;
    align-items: center;
  }
  .left{
    justify-content: flex-start;
    }
  .center{
    justify-content: center;
  }
  .right{
    align-items:flex-end;
  }
}

.default-no-runners {
  width: 710px;
  height: 100px;
  overflow: scroll;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding: 0 20px;
  box-sizing: border;
  border-top: 1px solid @bet-border-color;

  .noRunText {
    display: flex;
    font-family: Akrobat-SemiBold;
    font-size: 32px;
    color: @primary-font-color;
    margin-right: 30px;
    flex-shrink: 0;
  }

  .noRunner {
    display: flex;
    font-family: Akrobat-Regular;
    font-size: 32px;
    color: @remark-font-color;
    padding-right: 20px;
    box-sizing: border;
    flex-shrink: 0;
  }
}
