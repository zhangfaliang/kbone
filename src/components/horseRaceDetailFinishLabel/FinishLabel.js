import React from 'react';
import { HorseImg, HorseName, HorseNum } from '../horseRaceInfo';
import Ranking from './Ranking';
import styles from './index.less';

const FinishLabel = ({
  prefixCls = 'finish',
  odds = '5.6',
  favText = '',
  rankNum = '',
  rankingText = '',
  picUrl,
  shortName = '',
  horseCode = '',
  noEntry = '',
}) => {
  return (
    <div className={styles[`${prefixCls}-label`]}>
      <div className={styles.left}>
        {rankingText && <Ranking prefixCls={prefixCls} rankingText={rankingText} />}
        {picUrl !== undefined && <HorseImg prefixCls={prefixCls} picUrl={picUrl} />}
        {HorseNum && <HorseNum prefixCls={prefixCls} horseCode={horseCode} />}
        {HorseName && <HorseName prefixCls={prefixCls} shortName={shortName} />}
      </div>
      {rankNum && <span className={styles.center}>{rankNum}</span>}
      <div className={styles.right}>
        {odds && <span>{odds} </span>}
        {favText && <span>{favText}</span>}
        {noEntry && <span className={styles.noEntry}>{noEntry} </span>}
      </div>
    </div>
  );
};

FinishLabel.COMPONENT_NAME = 'FINISHLABEL';

export default FinishLabel;
