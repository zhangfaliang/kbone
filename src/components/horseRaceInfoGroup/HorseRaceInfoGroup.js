import React from 'react';
import HorseRaceInfoGroupTitle from './title';
import HorseRaceInfoCell from '../../components/horseRaceInfoCell';
import styles from './index.less';
import { NoRunners } from '../horseRaceDetailFinishLabel';

class HorseRaceInfoGroup extends React.Component {
  static COMPONENT_NAME = 'HORSERACEINFOGROUP';
  selectBetBtn = betInfo => {
    const {
      selectBetBtn,
      horseRaceType,
      sportName,
      leagueName,
      sportType,
      matchId,
      matchName,
      matchTime,
      matchExt,
      matchType,
    } = this.props;

    selectBetBtn &&
      selectBetBtn({
        ...betInfo,
        horseRaceType,
        sportName,
        leagueName,
        sportType,
        matchId,
        matchName,
        matchTime,
        matchExt,
        matchType,
      });
  };
  render() {
    const { children, prefixCls } = this.props;
    const childrens = React.Children.map(children, option => {
      const { ...other } = option.props;
      if (!option.type) {
        return '';
      } else if (option.type.COMPONENT_NAME === 'HORSERACEINFOGROUPTITLE') {
        return <HorseRaceInfoGroupTitle {...other} />;
      } else if (option.type.COMPONENT_NAME === 'HORSERACEINFOCELL') {
        return <HorseRaceInfoCell {...other} selectBetBtn={this.selectBetBtn} />;
      } else if (option.type.COMPONENT_NAME === 'NORUNNERS') {
        return <NoRunners {...other} />;
      }
    });
    const clsStr = styles[`${prefixCls}-horse-race-info-group`];
    return <div className={clsStr}>{childrens}</div>;
  }
}
HorseRaceInfoGroup.defaultProps = {
  prefixCls: 'default',
  selectBetBtn: betInfo => {
    console.log(betInfo);
  },
  horseRaceType: '',
  sportName: '',
  leagueName: '',
  sportType: '',
  matchId: '',
  matchName: '',
  matchTime: '',
  matchType: '',
};

export default HorseRaceInfoGroup;
