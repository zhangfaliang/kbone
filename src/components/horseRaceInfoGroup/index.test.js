import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import HorseRaceInfoGroup from './HorseRaceInfoGroup';
import Title from './title';
import HorseRaceInfoGroupTitle from './title';
import HorseRaceInfoCell from '../../components/horseRaceInfoCell';
import { NoRunners, FinishLabel, Ranking } from '../horseRaceDetailFinishLabel';
import HorseRaceBetButton from '../horseRaceBetButton';
import {
  SingleHorseInfo,
  ForeCastOrTriCast,
  SPInfo,
  HorseImg,
  HorseName,
  HorseNum,
} from '../../components/horseRaceInfo';
import HorseRaceBetButtonCell from '../horseRaceBetButtonCell';

describe('HorseRaceInfoGroup', async () => {
  it('HorseRaceInfoGroup', () => {
    const wrapper = render(
      <HorseRaceInfoGroup>
        <HorseRaceInfoGroupTitle showEmpty />
        <HorseRaceInfoCell>
          <SingleHorseInfo />
          <ForeCastOrTriCast />
          <SPInfo />
          <HorseRaceBetButtonCell>
            <HorseRaceBetButton showBetText isHide={false} historySp={[1, 3]} />
          </HorseRaceBetButtonCell>
          <FinishLabel>
            <HorseImg />
            <HorseName />
            <HorseNum />
            <Ranking />
          </FinishLabel>
        </HorseRaceInfoCell>
        <NoRunners noRunners={[1, 2, 3]} />
      </HorseRaceInfoGroup>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Title', () => {
    const wrapper = render(<Title />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HorseRaceInfoGroup', () => {
    const wrapper = mount(
      <HorseRaceInfoGroup>
        <HorseRaceInfoGroupTitle showEmpty />
        <HorseRaceInfoCell>
          <SingleHorseInfo />
          <ForeCastOrTriCast />
          <SPInfo />
          <HorseRaceBetButtonCell>
            <HorseRaceBetButton showBetText isHide={false} historySp={[1, 3]} />
          </HorseRaceBetButtonCell>
          <FinishLabel>
            <HorseImg />
            <HorseName />
            <HorseNum />
            <Ranking />
          </FinishLabel>
        </HorseRaceInfoCell>
        <NoRunners noRunners={[1, 2, 3]} />
      </HorseRaceInfoGroup>
    );
    const selectBetBtn = wrapper.prop('selectBetBtn');
    const click = jest.fn(selectBetBtn);
    const app = wrapper
      .find('HorseRaceInfoCell')
      .find('HorseRaceBetButtonCell')
      .find('HorseRaceBetButton')
      .simulate('click');
    expect(click()).toBe(undefined);
  });
});
