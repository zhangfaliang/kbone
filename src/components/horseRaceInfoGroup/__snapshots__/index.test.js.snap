// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorseRaceInfoGroup HorseRaceInfoGroup 1`] = `
<div
  class="default-horse-race-info-group"
>
  <div
    class="single-horse-race-info-group-title"
  >
    <span
      class="empty"
    />
    <span
      class="label"
    >
      info
    </span>
    <span
      class="text"
    />
  </div>
  <div
    class="single-horse-race-info-cell"
  >
    <div
      class="single-horse-info"
    >
      <div
        class="leftInfo"
      >
        <div
          class="top"
        >
          <div
            class="left"
          >
            <div
              class="single-horse-num"
            />
            <div
              class="single-horse-track-num"
            />
          </div>
          <img
            alt=""
            class="single-horse-img"
            src="[object Object]"
          />
        </div>
        <div
          class="single-horse-historical-record"
        />
      </div>
      <div
        class="rightInfo"
      >
        <div
          class="single-horse-name"
        />
        <div
          class="horseman"
        >
          <span />
           
          <span>
            / 
          </span>
          <span />
        </div>
        <div>
          <span
            class="age"
          >
             
          </span>
          <span
            class="weight"
          >
            100
          </span>
        </div>
      </div>
    </div>
    <div
      class="fore-or-tri-horse-info"
    >
      <div
        class="fore-or-tri-horse-num"
      />
      <div
        class="fore-or-tri-horse-name"
      />
    </div>
    <div
      class="default-sp-info"
    />
    <div
      class="default-horse-race-bet-btn-cell"
    >
      <div
        class="middle-horse-race-bet-btn"
      >
        <span
          class="betText"
        />
        <span
          class="odds"
        />
        <div
          class="historySp"
        >
          <span>
            1
          </span>
          <span>
            3
          </span>
        </div>
      </div>
    </div>
    <div
      class="finish-label"
    >
      <div
        class="left"
      >
        <div
          class="finish-horse-num"
        />
        <div
          class="finish-horse-name"
        />
      </div>
      <div
        class="right"
      >
        <span>
          5.6 
        </span>
      </div>
    </div>
  </div>
  <div
    class="default-no-runners"
  >
    <span
      class="noRunText"
    />
    <span
      class="noRunner"
    >
      1
    </span>
    <span
      class="noRunner"
    >
      2
    </span>
    <span
      class="noRunner"
    >
      3
    </span>
  </div>
</div>
`;

exports[`HorseRaceInfoGroup Title 1`] = `
<div
  class="single-horse-race-info-group-title"
>
  <span
    class="label"
  >
    info
  </span>
  <span
    class="text"
  />
</div>
`;
